src/main/java/Duke.java

```diff
@@ -108,6 +116,91 @@ public static void main(String[] args) {
         }
     }
 
+    // Handlers
+
+    private static <T extends Task> void OnNewTaskAdded(T t) {
+        taskList.add(t);
+        writeToFile(t.PopulateSaveData() + System.lineSeparator());
+        printNewTask(t);
+    }
+
+    private static void OnTaskRemoved(int index) {
+        PrintWithIndent("Noted. I've removed this task: ");
+        PrintWithIndent("  " + taskList.get(index));
+        taskList.remove(index);
+        PrintWithIndent("Now you have " + taskList.size() + " tasks in the list.");
+        try {
+            List<String> originalContent = Files.readAllLines(SavePath);
+            originalContent.remove(index);
+            Files.write(SavePath, originalContent);
+        } catch (IOException e) {
+            PrintWithIndent("Error occurs when removing a task: " + e);
+        }
+    }
+
+    private static void OnTaskDone(int index) {
+        Task task = taskList.get(index);
+        task.isDone = true;
+        PrintWithIndent("Nice! I've marked this task as done: ");
+        PrintWithIndent("  " + task);
+        try {
+            List<String> originalContent = Files.readAllLines(SavePath);
+            originalContent.set(index, task.toString());
+            Files.write(SavePath, originalContent);
+        } catch (IOException e) {
+            PrintWithIndent("Save error (mark task): " + e);
+        }
+    }
+
+
+    // I/O methods
+
+    private static void InitializeSaveData() {
+        try {
+            Path path = Arrays.stream(SaveDirectory)
```

Comment:
Change your method declaration to initializeSaveData. Try to follow the standard set by CS2103T in naming your functions